### PR TITLE
Add nicer logging/reporting to dev/update_packages.dart

### PR DESCRIPTION
**Pull request by eseidelGoogle**
Makes it more obvious where we are spending our time on Travis.

@Hixie

### CODE REVIEW

File: dev/update_packages.dart [hunk 1]
```diff
@@ -6,27 +6,39 @@
...
         workingDirectory: dir.path
       );
+      timer.stop();
+      stdout.write(" (${timer.elapsedMilliseconds} ms)");
       if (result.exitCode != 0) {
         print("... failed with exit code ${result.exitCode}.");
         print(result.stdout);
         print(result.stderr);
+      } else {
+        print(""); // Newline.
```
Comment (Hixie) [inline dev/update_packages.dart]: stdout.write("\n") for symmetry?

File: dev/update_packages.dart [hunk 2]
```diff
@@ -6,27 +6,39 @@
...
 }
 
 void main(List<String> arguments) {
+  Stopwatch timer = new Stopwatch()..start();
   bool upgrade = arguments.length > 0 && arguments[0] == '--upgrade';
   String FLUTTER_ROOT = new File(Platform.script.toFilePath()).parent.parent.path;
-  update(new Directory("$FLUTTER_ROOT/packages"), upgrade);
-  update(new Directory("$FLUTTER_ROOT/examples"), upgrade);
+  List<String> pubArgs = [ upgrade ? 'upgrade' : 'get' ];
+  int count = runPub(new Directory("$FLUTTER_ROOT/packages"), pubArgs);
```
Comment (Hixie) [inline dev/update_packages.dart]: this is a total nit, but for code prettiness I'd have `int count = 0;` followed by two parallel `+=` lines. :-)

**Comment by Hixie** [comment]
LGTM

File: dev/update_packages.dart [hunk 3]
```diff
@@ -6,27 +6,39 @@
...
       ProcessResult result = Process.runSync(
         binaryName,
-        [ upgrade ? 'upgrade' : 'get' ],
+        pubArgs,
         workingDirectory: dir.path
       );
+      timer.stop();
+      stdout.write(" (${timer.elapsedMilliseconds} ms)");
       if (result.exitCode != 0) {
         print("... failed with exit code ${result.exitCode}.");
```
Comment (Hixie) [inline dev/update_packages.dart]: there's more prints :-)

**Comment by eseidelGoogle** [comment]
@abarth, @chinmaygarde  presumably we could also use this travis.yml caching technology to cache the cloud storage assets, etc.

**Pull request merged**